includes/Core/Consent_Mode/REST_Consent_Mode_Controller.php

```diff
@@ -132,6 +132,52 @@ protected function get_rest_routes() {
 					),
 				)
 			),
+			new REST_Route(
+				'core/site/data/consent-api-info',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function () {
+							$is_active = function_exists( 'wp_set_consent' );
+							$installed = $is_active;
+							$slug      = 'wp-consent-api';
+							$plugin    = "$slug/$slug.php";
+
+							$response = array(
+								'hasConsentAPI' => $is_active,
+							);
+
+							if ( ! $is_active ) {
+								if ( ! function_exists( 'get_plugins' ) ) {
+									require_once ABSPATH . 'wp-admin/includes/plugin.php';
+								}
+								foreach ( array_keys( get_plugins() ) as $installed_plugin ) {
+									if ( $installed_plugin === $plugin ) {
+										$installed = true;
+										break;
+									}
+								}
+
+								// Alternate wp_nonce_url without esc_html breaking query parameters.
+								$nonce_url = function ( $action_url, $action ) {
+									return add_query_arg( '_wpnonce', wp_create_nonce( $action ), $action_url );
+								};
+								$activate_url = $nonce_url( self_admin_url( 'plugins.php?action=activate&plugin=' . $plugin ), 'activate-plugin_' . $plugin );
+								$install_url = $nonce_url( self_admin_url( 'update.php?action=install-plugin&plugin=' . $slug ), 'install-plugin_' . $slug );
+
+								$response['wpConsentPlugin'] = array(
+									'installed'   => $installed,
```

Comment:
Note that there's no need to include the `active` field specified in the AC. This is because `hasConsentAPI` indicates the active status.